examples/energy-management-app/energy-management-common/src/EVSEManufacturerImpl.cpp

```diff
@@ -89,6 +93,182 @@ CHIP_ERROR EVSEManufacturer::Shutdown()
     return CHIP_NO_ERROR;
 }
 
+/**
+ * @brief   Allows a client application to send in power readings into the system
+ *
+ * @param[in]  aEndpointId     - Endpoint to send to EPM Cluster
+ * @param[in]  aActivePower_mW - Power measured in milli-watts
+ * @param[in]  aVoltage_mV     - Voltage measured in milli-volts
+ * @param[in]  aCurrent_mA     - Current measured in milli-amps
+ */
+CHIP_ERROR EVSEManufacturer::SendPowerReading(EndpointId aEndpointId, int64_t aActivePower_mW, int64_t aVoltage_mV,
+                                              int64_t aCurrent_mA)
+{
+    // TODO add Power Readings when EPM cluster is merged
+
+    return CHIP_NO_ERROR;
+}
+
+/**
+ * @brief   Allows a client application to send in energy readings into the system
+ *
+ *          This is a helper function to add timestamps to the readings
+ *
+ * @param[in]  aCumulativeEnergyImported -total energy imported in milli-watthours
+ * @param[in]  aCumulativeEnergyExported -total energy exported in milli-watthours
+ */
+CHIP_ERROR EVSEManufacturer::SendEnergyReading(EndpointId aEndpointId, int64_t aCumulativeEnergyImported,
+                                               int64_t aCumulativeEnergyExported)
+{
+    MeasurementData * data = MeasurementDataForEndpoint(aEndpointId);
+
+    EnergyMeasurementStruct::Type energyImported;
+    EnergyMeasurementStruct::Type energyExported;
+
+    uint32_t oldTimestamp;
+
+    // Get current timestamp
+    uint32_t currentTimestamp;
+    CHIP_ERROR err = GetEpochTS(currentTimestamp);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(AppServer, "GetEpochTS returned error getting timestamp");
+        return err;
+    }
+
+    /** IMPORT */
+    // Copy last endTimestamp into new startTimestamp if it exists
+    energyImported.startTimestamp.ClearValue();
+    if (data->cumulativeImported.HasValue())
+    {
+        if (data->cumulativeImported.Value().endTimestamp.HasValue())
```

Comment:
could we directly do:

```cpp
energyImported.startTimestamp = data->cumulativeImported.Value().endTimestamp;
```
?

Thinking that startTimestamp is cleared ... and it seems we want to take endTimeStamp if it has some value. If it does not, it stays cleared. Same question for export below.

Reply:
Done